### PR TITLE
Introduce SearchAssets service

### LINKED ISSUE #1502: Add asset filter to category grid
Asset filter should be present for category grid on `admin/media_gallery_catalog/category/index` page

Depends on https://github.com/magento/adobe-stock-integration/issues/1481

### PR DESCRIPTION
**Pull request by Nazar65**
### Description (*)

This service is responsible for search media assets by searchCriteria filters

### Fixed Issues (if relevant)

1. Fixes magento/adobe-stock-integration#1502: add asset filter to category grid

### Manual testing scenarios (*)
1.Open Category grid page by link `admin/media_gallery_catalog/category/index`
2. Expand Filters tab
3. Start typing asset titile in Asset filter input

Screenshot_2020-07-10 Categories Magento Admin

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by Nazar65** [comment]
@magento run all tests

File: app/code/Magento/MediaGalleryApi/Api/SearchAssetsInterface.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * Search media gallery assets by search criteria
+ * @api
+ */
+interface SearchAssetsInterface
+{
+    /**
+     * Search media gallery assets
+     *
+     * @param SearchCriteriaInterface $searchCriteria
+     * @return AssetInterface[]
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryApi/Api/SearchAssetsInterface.php]: let's return the assets search result interface for the consistency with other search services

File: app/code/Magento/MediaGalleryApi/Api/SearchAssetsInterface.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+namespace Magento\MediaGalleryApi\Api;
+
+use Magento\Framework\Api\Search\SearchResultInterface;
+use Magento\Framework\Api\SearchCriteriaInterface;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\MediaGalleryApi\Api\Data\AssetInterface;
+
+/**
+ * Search media gallery assets by search criteria
+ * @api
```
Comment (sivaschenko) [inline app/code/Magento/MediaGalleryApi/Api/SearchAssetsInterface.php]: please remove the api annotation

File: app/code/Magento/MediaGallery/Model/ResourceModel/SearchAssets.php [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        return $assets;
+    }
+
+    /**
+     * Retrieve assets data from database
+     *
+     * @param SearchCriteriaInterface $searchCriteria
+     * @return SearchResultInterface
+     */
+    private function getAssetsData(SearchCriteriaInterface $searchCriteria): SearchResultInterface
```
Comment (sivaschenko) [inline app/code/Magento/MediaGallery/Model/ResourceModel/SearchAssets.php]: please extract SeachCriteria processing to a separate ResourceModel class (while the main service implementation class can be in the Model namespace)

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the pull request @Nazar65 ! Please see my comments

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by Nazar65** [comment]
@magento run all tests

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-7827](https://jira.corp.magento.com/browse/ENGCOM-7827) has been created to process this Pull Request
:eight_spoked_asterisk: @sivaschenko, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by lbajsarowicz** [comment]
Nice feature, @Nazar65 :clap:

**Comment by Nazar65** [comment]
@magento run all tests

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

Test scenario on cucumber studio - https://studio.cucumber.io/projects/131313/test-plan/folders/1320712/scenarios/4951024

**Pull request merged**